src/pyats/contrib/creators/topology.py

```diff
@@ -0,0 +1,656 @@
+        try:
+            lldp = device.api.get_lldp_neighbors_info()
+        except:
+            log.error("Exception occurred getting lldp info", exc_info = True)
+        return {device.name: {'cdp':cdp, 'lldp':lldp}}
+
+    def get_device_connections(self, data, entry, device_list, ip_net, int_skip, testbed):
+        '''
+        Take a device from a testbed and find all the adjacent devices
+        '''
+        connection_dict = {}
+        # filter to strip out the domain name from the system name
+        domain_filter = re.compile(r'^.*?(?P<hostname>[-\w]+)\s?')
+
+        # get and parse cdp information
+        result = data['cdp']
+        log.info('cdp neighbor information: {}'.format(result))
+        if result is not None:
+            # for every cdp entry find the os, destination name, destination 
+            # port, and the ip-address and then add them to the relevant lists
+            for index in result['index']:
+                # get the relevant information
+                connection = result['index'][index]
+                dest_host = connection.get('system_name')
+                if not dest_host:
+                    dest_host = connection.get('device_id')
+                m = domain_filter.match(dest_host)
+                if m:
+                    dest_host = m.groupdict()['hostname']
+                if self._topo_only and dest_host not in testbed.devices:
+                    log.info('{} is not in initial testbed, skipping connection'.format(dest_host))
+                    continue
+
+                dest_port = connection['port_id']
+                interface = connection['local_interface']
+                # if either the local or neigboring interface is in ignore list
+                # do not log the connection on move on to the next one
+                log.info('interface = {interface},'
+                         'dest = {dest}'.format(interface = interface, 
+                                                dest = dest_port))
+                if interface in int_skip or dest_port in int_skip:
+                    log.info('connection or dest interface is found in' 
+                            'ignore interface list, skipping connection')
+                    continue
+                ip_set = set ()
+                # get the ip addresses for the neighboring device
+                x = connection.get('management_addresses')
+                if x is not None:
+                    for address in x:
+                        ip_set.add(address)
+                
+                x = connection.get('entry_addresses')
+                if x is not None:
+                    for address in x:
+                        ip_set.add(address)
+                skip = False
+                # if the ip addresses for the connection are in the range given
+                # by the cli, do not log the connection and move on
+                for ip in ip_set:

```

Comment:
I think scope of discovery is controlled by `exclude_network`. and here just trying to avoid adding same one? so looks good to me.

@chowed85 please confirm

Reply:
Yeah this is so that if the connection is in a network that the user wanted to exclude the script will just not log the connection and skip over it